src/adapter/base.cr

```diff
@@ -60,6 +60,15 @@ abstract class Granite::Adapter::Base
     Granite::Adapter::Base.env(url)
   end
 
+  # Use macro in order to read a constant defined in each subclasses.
+  macro inherited
+    # quotes table and column names
+    def quote(name : String) : String
+      char = QUOTING_CHAR
+      char + name.gsub(char, "#{char}#{char}") + char
```

Comment:
I'm a little concerned with this logic and performance.  If someone already quoted a query:
```
select `table`.`column` from `table`;
```
what will this produce?

Also, do we need to perform a `gsub` for every table and column name or can we improve this?

Reply:
Strictly speaking, it will be slightly slow, but `gsub` is fast and the target string is short so we can ignore the influence on the whole. From the performance point of view, I think there are many other places that can speed up. Furthermore, the implementation by **Crystal** is already too fast enough.

Therefore, unless it is a clear problem at the present time, I would like to prioritize "features" over "performance". Yep, I'd like to implement many features like following! 😺 
- support natural keys
- bang method like `save!`, `find!`
- avoid ugly and redundant code like `record.name.not_nil!`

Reply:
If this was called one time, I would agree but its getting called multiple times, once per each column.  I agree that the feature outweighs the cost.  I was wondering if there was a better way to do this.  I'm ok with this for now.  We can look at optimizing it later.

Thanks for your contribution.

Reply:
@drujensen hopefully granite can defer to lower level database specific adapters at some point, I'd rather not own the code that does escaping 😰

Reply:
@robacarp right, right.  I am not completely comfortable with this change.  I know we need it but something feels odd here.